### PR TITLE
Use redux

### PR DESCRIPTION
**Pull request by yuvenalmash**
## changes summary
- connect app to redux store
- display list of books from redux store
- use useDispatch to call action creators
- use useSelector to get data from redux store

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -27,18 +44,14 @@ const bookReducer = (state = initialState, action) => {
...
-    });
-};
+export const addBook = (bookObj) => (
+  {
+    type: ADD_BOOK,
+    id: bookObj.id,
+    title: bookObj.title,
+    author: bookObj.author,
+    genre: bookObj.genre,
+  });
```
Comment (anagudelogu) [inline src/redux/books/books.js]: - [x] **[OPTIONAL]:** You did great working on this milestone! 💯 However, if you want to improve your code, you could follow a widely used convention for action creators: the `payload` key to send the data to the reducer, like this:

```suggestion
export const addBook = (bookObj) => (
  {
    type: ADD_BOOK,
    payload: bookObj
  });
```

This way, you don't have to list all object's properties, and in your reducer, you could implement the following change too:

```js
    case ADD_BOOK:
      return [
        ...state,
        {
          ...action.payload,
          completed: '0%',
          currentChapter: '0',
        },
      ];

```

Requiring less lines of code and making your code more readable in the long run! 🤓 🔧

File: src/components/CategoriesContainer.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import React, { useState } from 'react';
-import store from '../redux/configureStore';
+import { useDispatch, useSelector } from 'react-redux';
+import { checkStatus } from '../redux/categories/categories';
 
 const CategoriesContainer = () => {
   const [dynamicContent, setDynamicContent] = useState([]);
-  const categories = store.getState().categories[0];
+  const dispatch = useDispatch();
+  const status = useSelector((state) => state.categories);
```
Comment (anagudelogu) [inline src/components/CategoriesContainer.js]: - [x] Please make sure that when the user clicks the `Check status` button, the message `"Under construction"` is displayed. At the moment, I have to click twice to get this message. Kindly look into it and fix it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @yuvenalmash , 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @yuvenalmash, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**